resources/js/services.js

```diff
@@ -0,0 +1,156 @@
+import utils from "./utils.js"
+
+// Global variables
+let contentObj, headerObj
+
+/*--------------- Services page content loading section ---------------*/
+utils.jsonCaller("get", "resources/json/services.json",function(object){
+  headerObj = object.table
+  loadTable(headerObj)
+})
+
+
+/*--------------- Table loading section ---------------*/
+function loadTable(headerObj){
+  utils.jsonCaller("get", "resources/json/offices.json",function(object){
+    contentObj = object
+    tableLoader(headerObj,contentObj,'asc')
+  })
+}
+
+function tableLoader(headerObj,contentObj,order){
+  const tableHeader = document.querySelector("#table-head")
+  const tableBody = document.querySelector("#table-body")
+
+  loadTableHead(tableHeader, headerObj,order)
+
+   for (let contentValue of contentObj) {
+     let rowElement = document.createElement("tr");
+     for (let headerValue of headerObj) {
+       let columnElement = document.createElement("td");
+       let key = headerValue.id;
+       let cellValue = contentValue[key];
+       if (headerValue.type == "link") {
+         let link = document.createElement("a");
+         link.href = cellValue;
+         link.textContent = "Link";
+         columnElement.appendChild(link);
+       } else if (headerValue.type == "button") {
+         if (contentValue.status == "open") {
+           let btn = document.createElement("button");
+           btn.textContent = "Apply Now";
+           columnElement.appendChild(btn);
+         } else {
+           columnElement.textContent = " - ";
+         }
+       } else if (headerValue.type == "number" || headerValue.type == "date") {
+         columnElement.style.textAlign = "right";
```

Comment:
What if I wanted to right-align a particular text/link column values?

Better to listen to a new column-config-key to rightAlign: true/false, and append class to table cells accordingly right?

Reply:
This particular section of code is only to make sure the table follows universal method of data alignment. User has no power to change the alignment later.

Reply:
Ok